enhancements/workload-partitioning/management-workload-partitioning-api.md

```diff
@@ -0,0 +1,249 @@
+   clearly communicate how an enhancement is coded up and delivered. If an
+   enhancement describes a new deployment topology or platform, include a
+   logical description for the deployment, and how it handles the unique aspects
+   of the platform. Aim for single topic PRs to keep discussions focused. If you
+   disagree with what is already in a document, open a new PR with suggested
+   changes.
+1. **Keep all required headers.** If a section does not apply to an
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+The `Metadata` section above is intended to support the creation of tooling
+around the enhancement process.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The overall feature summary is present in the [Management workload partitioning](workload-partitioning/management-workload-partitioning.md) feature. This document focuses on how the feature enablement API and its interactions will look like.
+
+## Motivation
+
+This feature is about designing a read-only API that will describe the enabled workload partitions (types, classes, etc.). This information is needed for kubelet to start exposing the right resources as well for the admission webhook to know when the pod manipulation is needed.
+
+### Goals
+
+The short term goal is to get enough functionality to create the `management` partition where all OpenShift infrastructure could be placed. This partition is cluster wide and only affects pods with certain pre-agreed-upon labels and placed in specific pre-annotated namespaces.
+
+### Non-Goals
+
+- Long term goal of allowing cluster admins to create their own partitions is not part of this proposal for now
+- Day 2 creation of partitions is not part of the proposal
+- Partitions with scope less than the entire cluster (such as being limited to Namespaces or MCPs) are not part of the proposal
+
+## Proposal
+
+The proposal is to define a new cluster-wide Custom Resource Definition that would describe the allowed partition names in the status section. That way it hints at being a read only object where no user/admin input or modifications are expected.
+
+```yaml
+apiVersion: workload.openshift.io/v1
+kind: WorkloadPartitions
+metadata:
+  # arbitrary name, all objects of this Kind should be processed and merged
+  name: management-partition
+status:
+  # List of strings, defines partition names that will be recognized by the
+  # workload partitioning webhook. This list will also inform PAO about partitions
+  # that should be configured on the kubelet and CRI-O level.
+  clusterPartitionNames:
```

Comment:
hm what about CPUs that should be used for the CPU pinning or will it use reservedCPUs by default?

Reply:
The API server does not care. Kubelet / CRIO will get the right ids from PAO.